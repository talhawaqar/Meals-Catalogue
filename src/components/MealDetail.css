.hero {
  height: 400px;
  background-size: cover;
}

.main {
  background-color: #fafafa;
}

.main h1 {
  font-size: 32px;
}

.main span.heading {
  font-weight: bold;
  font-size: 16px;
}
