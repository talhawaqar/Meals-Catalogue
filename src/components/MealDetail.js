import React from 'react';
import { connect } from 'react-redux';
import { fetchMeal } from '../actions';
import './MealDetail.css';

class MealDetail extends React.Component {
  
  componentDidMount(){
    this.props.fetchMeal(this.props.match.params.id);
  }

  renderIngrediants = () => {
    const meal = this.props.meal;
    let ingrediants = [<span> 1. { meal.strIngredient1 } </span>, <span> { meal.strMeasure1 } </span>, <br/>];
    for( let i=2; i<=20; i++ ) {
      if(meal[`strIngredient${i}`] === '')
      {
        break;
      }
      ingrediants.push(<span> {`${i}`}. { meal[`strIngredient${i}`] } </span>);
      ingrediants.push(<span> { meal[`strMeasure${i}`] } </span>);
      ingrediants.push(<br/>);  
    }
    return ingrediants;
  }

  render() {
    if(!this.props.meal) {
      return <div>Loading....</div>;
    }

    const meal = this.props.meal;
    return (
      <>
        <div className='container main'>
          <div style = {{ backgroundImage: `url(${meal.strMealThumb})` }}className='hero w-100'>
          </div> 
          <h1 className='mt-3' > { meal.strMeal } </h1>
          <div className='row mt-4'>
            <div className='col-12'>
              <span className='heading' >Category:</span> <span className='detail'> { meal.strCategory } </span>
              <br/>
              <span className='heading' >Ingredients:</span>
              <br/>
              { this.renderIngrediants() }
              <span className='heading' >Cooking Instructions:</span>
              <br/>
              <p>
                { meal.strInstructions }
              </p>
            </div>
          </div>
        </div>
      </>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    meal: state.meals[ownProps.match.params.id]
  };
}

export default connect( mapStateToProps, { fetchMeal } )(MealDetail);
