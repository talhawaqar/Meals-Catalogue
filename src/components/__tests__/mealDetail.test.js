import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import { MemoryRouter } from 'react-router-dom';
import reduxThunk from 'redux-thunk';
import reducer from '../../reducers';
import MealDetail from '../MealDetail';

it('renders correctly when store is provided', () => {
  const mealId = '52772';
  const match = { params: { id: '52772' } };
  const store = createStore(reducer, applyMiddleware(reduxThunk));
  const mealDetail = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={[`/meals/${mealId}`]}>
        <MealDetail match={match} />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(mealDetail).toMatchSnapshot();
});
