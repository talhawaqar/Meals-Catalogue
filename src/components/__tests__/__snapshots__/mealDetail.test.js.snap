// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when store is provided 1`] = `
<div>
  Loading....
</div>
`;
